vault/namespaces_oss.go

```diff
@@ -38,3 +39,22 @@ func NamespaceView(barrier logical.Storage, ns *namespace.Namespace) BarrierView
 func (c *Core) NamespaceByPath(ctx context.Context, path string) (*namespace.Namespace, string) {
 	return c.namespaceStore.GetNamespaceByLongestPrefix(ctx, path)
 }
+
+func (c *Core) NamespaceByStoragePath(ctx context.Context, path string) (*namespace.Namespace, string, error) {
```

Comment:
Just since I'm picky about this, let's move this to `namespace_store.go`, without the HashiCorp copyright header.